File: 0131/led_game.py
import RPi.GPIO as GPIO
import random
import time

# GPIO 핀 번호 설정
LED_RED = 4
LED_GREEN = 5
LED_BLUE = 6

# GPIO 핀 모드 설정
GPIO.setmode(GPIO.BCM)
GPIO.setup(LED_RED, GPIO.OUT)
GPIO.setup(LED_GREEN, GPIO.OUT)
GPIO.setup(LED_BLUE, GPIO.OUT)

# LED 초기화
GPIO.output(LED_RED, GPIO.LOW)
GPIO.output(LED_GREEN, GPIO.LOW)
GPIO.output(LED_BLUE, GPIO.LOW)

# 게임 시작
print("------------------------------------------------")
print("                빛의 삼원색 게임 ")
print("- 색을 확인하고 합쳐지면 어떤 색이 되는지 맞춰보자 -")
print("------------------------------------------------\n")

# 랜덤 색 2개 or 3개 선정
randColor = [0, 0, 0]
if random.randint(0, 2) != 0:
    # 화이트 이외의 다른 색
    randColor[0] = random.randint(1, 3)
    randColor[1] = random.randint(1, 3)
    while randColor[0] == randColor[1]:
        randColor[1] = random.randint(1, 3)
else:
    # 화이트가 되는 경우
    randColor[2] = 7

# 랜덤 색 출력
time.sleep(1)
if randColor[2] == 7:
    GPIO.output(LED_RED, GPIO.HIGH)
    time.sleep(0.5)
    GPIO.output(LED_RED, GPIO.LOW)
    GPIO.output(LED_GREEN, GPIO.HIGH)
    time.sleep(0.5)
    GPIO.output(LED_GREEN, GPIO.LOW)
    GPIO.output(LED_BLUE, GPIO.HIGH)
    time.sleep(0.5)
    GPIO.output(LED_BLUE, GPIO.LOW)
else:
    # 다른 색인 경우
    if randColor[0] == 1:
        GPIO.output(LED_RED, GPIO.HIGH)
        time.sleep(0.5)
        GPIO.output(LED_RED, GPIO.LOW)
    elif randColor[0] == 2:
        GPIO.output(LED_GREEN, GPIO.HIGH)
        time.sleep(0.5)
        GPIO.output(LED_GREEN, GPIO.LOW)
    elif randColor[0] == 3:
        GPIO.output(LED_BLUE, GPIO.HIGH)
        time.sleep(0.5)
        GPIO.output(LED_BLUE, GPIO.LOW)

    if randColor[1] == 1:
        GPIO.output(LED_RED, GPIO.HIGH)
        time.sleep(0.5)
        GPIO.output(LED_RED, GPIO.LOW)
    elif randColor[1] == 2:
        GPIO.output(LED_GREEN, GPIO.HIGH)
        time.sleep(0.5)
        GPIO.output(LED_GREEN, GPIO.LOW)
    elif randColor[1] == 3:
        GPIO.output(LED_BLUE, GPIO.HIGH)
        time.sleep(0.5)
        GPIO.output(LED_BLUE, GPIO.LOW)

# 선택 옵션 출력
print("---------------------------")
print("1. yellow")
print("2. magenta")
print("3. cyan")
print("4. white")
print("---------------------------\n")

# 답 입력 받기
number = int(input("숫자를 입력하세요: "))

# 정답일 경우
if number + 2 == randColor[0] + randColor[1] or number + 3 == randColor[2]:
    print("정답입니다.")
    GPIO.output(LED_RED, GPIO.HIGH)
    time.sleep(0.5)
    GPIO.output(LED_GREEN, GPIO.HIGH)
    time.sleep(0.5)
    GPIO.output(LED_BLUE, GPIO.HIGH)
    time.sleep(0.5)
    GPIO.output(LED_RED, GPIO.LOW)
    GPIO.output(LED_GREEN, GPIO.LOW)
    GPIO.output(LED_BLUE, GPIO.LOW)
else:
    # 오답일 경우
    print("오답입니다.")
    GPIO.output(LED_RED, GPIO.HIGH)
    time.sleep(0.5)
    GPIO.output(LED_RED, GPIO.LOW)
    time.sleep(0.5)
    GPIO.output(LED_RED, GPIO.HIGH)
    time.sleep(0.5)
    GPIO.output(LED_RED, GPIO.LOW)
    time.sleep(0.5)
    GPIO.output(LED_RED, GPIO.HIGH)
    time.sleep(0.5)
    GPIO.output(LED_RED, GPIO.LOW)

GPIO.cleanup()

File: 0131/led_game.c
#include <stdio.h>
#include <stdlib.h>
#include <wiringPi.h>
#include <unistd.h>
#include <time.h>

#define LED_RED 7
#define LED_GREEN 21
#define LED_BLUE 22

int main(void){
    srand(time(NULL));
    int randColor[3] = {0, 0, 0};
    int number;

    //핀 모드 설정
	if(wiringPiSetup () == -1)
		return 1;
	pinMode(LED_RED,OUTPUT);
	pinMode(LED_GREEN,OUTPUT);
	pinMode(LED_BLUE,OUTPUT);

    //LED 초기화
	digitalWrite(LED_RED,0);
	digitalWrite(LED_GREEN,0);
	digitalWrite(LED_BLUE,0);
	
    //게임 시작
    printf("------------------------------------------------\n");
    printf("                빛의 삼원색 게임 \n");
    printf("- 색을 확인하고 합쳐지면 어떤 색이 되는지 맞춰보자 -\n");
	printf("------------------------------------------------\n\n");

    //랜덤 색 2개 or 3개 선정 
    if (rand()%3 != 0){
        //화이트 이외의 다른 색
        randColor[0] = rand() % 3 + 1;
        randColor[1] = rand() % 3 + 1;
        while (randColor[0] == randColor[1]){
            randColor[1] = rand() % 3 + 1;
        }
    }else{
        //화이트가 되는 경우
        randColor[2] = 7;
    }

    //랜덤 색 출력
    //화이트가 되는 경우
    usleep(1000000);
    if(randColor[2] == 7){
        digitalWrite(LED_RED,1);
	    usleep(500000);
	    digitalWrite(LED_RED,0);
        digitalWrite(LED_GREEN,1);
	    usleep(500000);
	    digitalWrite(LED_GREEN,0);
        digitalWrite(LED_BLUE,1);
	    usleep(500000);
	    digitalWrite(LED_BLUE,0);
    }else{
        //다른 색인 경우 
        switch(randColor[0]){
            case 1:
                digitalWrite(LED_RED,1);
                usleep(500000);
                digitalWrite(LED_RED,0);
                break;
            case 2:
                digitalWrite(LED_GREEN,1);
                usleep(500000);
                digitalWrite(LED_GREEN,0);
                break;
            case 3:
                digitalWrite(LED_BLUE,1);
                usleep(500000);
                digitalWrite(LED_BLUE,0);
                break;
            default:
                printf("출력 오류");
                break;    
        }

        switch(randColor[1]){
            case 1:
                digitalWrite(LED_RED,1);
                usleep(500000);
                digitalWrite(LED_RED,0);
                break;
            case 2:
                digitalWrite(LED_GREEN,1);
                usleep(500000);
                digitalWrite(LED_GREEN,0);
                break;
            case 3:
                digitalWrite(LED_BLUE,1);
                usleep(500000);
                digitalWrite(LED_BLUE,0);
                break;
            default:
                printf("출력 오류");
                break;    
        }
    }

    //선택 옵션 출력
    printf("---------------------------\n");
    printf("1. yellow\n");
    printf("2. magenta\n");
    printf("3. cyan\n");
    printf("4. white\n");
    printf("---------------------------\n\n");

    //답 입력 받기 
    printf("숫자를 입력하세요: ");
    scanf("%d", &number);

    //정답일 경우
    if (number+2 == randColor[0]+randColor[1] || number+3 == randColor[2]){ 
        printf("정답입니다.\n");
        digitalWrite(LED_RED,1);
	    usleep(500000);
        digitalWrite(LED_GREEN,1);
	    usleep(500000);
        digitalWrite(LED_BLUE,1);
	    usleep(500000);
        digitalWrite(LED_RED,0);
        digitalWrite(LED_GREEN,0);
	    digitalWrite(LED_BLUE,0);
    }else{
        //오답일 경우 
        printf("오답입니다.\n");
        digitalWrite(LED_RED,1);
	    usleep(500000);
	    digitalWrite(LED_RED,0);
        usleep(500000);
        digitalWrite(LED_RED,1);
	    usleep(500000);
	    digitalWrite(LED_RED,0);
        usleep(500000);
        digitalWrite(LED_RED,1);
	    usleep(500000);
	    digitalWrite(LED_RED,0);
    }

	return 0;
}